### PR TITLE
Interleave optimized.

### PR DESCRIPTION
**Pull request by coady**
About 2x faster from minimizing the inner loop.  Also cleaned up exception handling.

### CODE REVIEW

**Comment by coady** [comment]
In []: seqs = [xrange(10 ** n) for n in range(7)]

In []: %timeit deque(interleave(seqs), 0)
1 loop, best of 3: 335 ms per loop

In []: %timeit deque(interleave2(seqs), 0)
10 loops, best of 3: 155 ms per loop

**Comment by eriknw** [comment]
Hooray for optimization!

File: toolz/itertoolz.py [hunk 1]
```diff
@@ -214,7 +214,7 @@ def _merge_sorted_binary_key(seqs, key):
         yield val1
 
 
-def interleave(seqs, pass_exceptions=()):
+def interleave(seqs):
```
Comment (eriknw) [inline toolz/itertoolz.py]: Why did you remove `pass_exceptions=()` argument?

Comment (coady) [inline toolz/itertoolz.py]: Well, it didn't appear to be documented or tested, so hopefully backwards compatibility isn't an issue.

Also, I don't get the use case.  Iterables must support the iteration protocol.  If custom substitutes for `StopIteration` are supported, wouldn't that affect everything in `itertoolz`?

Comment (eriknw) [inline toolz/itertoolz.py]: @mrocklin, care to chime in?  I believe `pass_exceptions=()` was included from the very beginning when you consolidated your "utils.py" files from various projects. I know `logpy`, for instance, uses `pass_exceptions` in the `interleave` function it defines.

I agree that `pass_exceptions` seems out of place and not sufficiently justified, but @mrocklin has needed it in the past, and I'm generally in favor of maintaining backwards compatibility.

Comment (mrocklin) [inline toolz/itertoolz.py]: No clue. +1 to dropping it.

Comment (eriknw) [inline toolz/itertoolz.py]: Great!  Let's keep it out.

**Comment by llllllllll** [comment]
Anythin blocking this?
